src/cb_admin/pages/ConfirmPassword.js

```diff
@@ -25,24 +26,33 @@ export default class ConfirmPassword extends React.Component {
 
     this.state = {
       errors: {},
+      email: null,
     };
   }
 
+  componentDidMount() {
+    CbAdmin.get()
+      .then(res => this.setState({ email: ResponseUtils.getResponse(res).email }))
+      .catch(error => redirectOnError(this.props.history.push, error, { 403: '/admin/confirm' }));
```

Comment:
If there's an error throw from `then`, this introduces a kind of infinite loop. You probably want to forward `/confirm` if the response is >= `400`, but otherwise forward to home or something.